src/GafferArnold/ArnoldShader.cpp

```diff
@@ -59,12 +61,15 @@ using namespace IECore;
 using namespace GafferScene;
 using namespace GafferArnold;
 using namespace Gaffer;
+using namespace GafferOSL;
 
 GAFFER_GRAPHCOMPONENT_DEFINE_TYPE( ArnoldShader );
 
 ArnoldShader::ArnoldShader( const std::string &name )
 	:	GafferScene::Shader( name )
 {
+	// This is to allow Arnold Shaders to be connected to OSL Shaders
+	OSLShader::registerCompatibleShader( staticTypeId() );
```

Comment:
Ideally we'd be calling this statically somehow rather than every time a shader was constructed. One simple approach might be to have `registerCompatibleShader()` return a bool for success/failure and then do this?

`static bool g_oslRegistration = OSLShader::registerCompatibleShader( staticTypeId() );`

Reply:
Sounds good, so I should do something like this in the constructor: 
```
if( !g_oslRegistration )
    static bool g_oslRegistration = OSLShader::registerCompatibleShader( staticTypeId() );
```

Also I can probably const correct the argument to IECore::TypeId (or an IECore::InternedString for "ai:surface").

Reply:
Nevermind I get you now, so that I can run it in an anonymous namespace and skip the constructor entirely.